src/sql/azdata.proposed.d.ts

```diff
@@ -385,6 +385,29 @@ declare module 'azdata' {
 		title: string;
 	}
 
+	export interface ConnectionProvider extends DataProvider {
+		/**
+		 * Changes the password for a SQL Server connection profile.
+		 */
+		changePassword(connectionUri: string, connectionInfo: ConnectionInfo, newPassword: string): Thenable<PasswordChangeResult>;
+	}
+
+	// Password Change Request ----------------------------------------------------------------------
+	export interface PasswordChangeResult {
+		/**
+		 * Whether the password change was successful
+		 */
+		result: boolean;
+		/**
+		 * The error message if the password change was unsuccessful
+		 */
+		messages?: string | undefined;
```

Comment:
This seems like it should be `errorMessage` and then the one below `errorDetails` or something given the comments